cpp/8puzzle.cpp

```diff
@@ -335,6 +337,19 @@ bool PuzzleState::IsSameState( PuzzleState &rhs )
 
 }
 
+// The 9 tiles positions can be encoded as digits
+size_t PuzzleState::Hash()
+{
+	stringstream stream;
+	for( size_t i = 0; i < (BOARD_HEIGHT * BOARD_WIDTH); i++ )
+	{
+		stream << tiles[i];
+	}
+	size_t state_digits;

```

Comment:
I may be misunderstanding but I think this should be more like:
  return std::hash<std::string>{}(stream.str());

Reply:
Each state is defined by an ordering of the tiles. Turning them into a number is a unique hash because it would be another state if it shared the same ordering of digits. The has of a size_t is just identity function I believe. Turning the string into a hash would also work but seemed less intuitive. I'm fine either way.

Reply:
You are right. I chose to refactor it a little but it is now somewhere between your solution and mine